oap-server/server-receiver-plugin/otel-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/otel/OtelMetricReceiverProvider.java

```diff
@@ -61,21 +64,33 @@ public void onInitialized(final OtelMetricReceiverConfig initialized) {
 
     @Override
     public void prepare() throws ServiceNotProvidedException, ModuleStartException {
+        if (config.getEnabledHandlers().isEmpty()) {
+            return;
+        }
+        this.handlers =
+            Handler.all().stream()
+                   .filter(h -> config.getEnabledHandlers().contains(h.type()))
+                   .peek(h -> {
+                       if (h instanceof OpenTelemetryMetricRequestProcessor) {
+                           registerServiceImplementation(
+                               OpenTelemetryMetricRequestProcessor.class,
+                               (OpenTelemetryMetricRequestProcessor) h
+                           );
+                       }
```

Comment:
This is a wrong implementation, `registerServiceImplementation` will override an existing service implementation quietly, it might work for now, but when we add new classes that implement `OpenTelemetryMetricRequestProcessor` some day in the future, you might get a wrong implementation with 


```java
final OpenTelemetryMetricRequestProcessor processor =
    getManager().find(OtelMetricReceiverModule.NAME)
                .provider()
                .getService(OpenTelemetryMetricRequestProcessor.class);
```